var/spack/repos/builtin/packages/rocm-smi-lib/package.py

```diff
@@ -119,3 +130,32 @@ def post_install(self):
             shutil.rmtree(self.prefix.rocm_smi)
             os.remove(join_path(self.prefix.bin, "rsmiBindings.py"))
             symlink("../bindings/rsmiBindings.py", join_path(self.prefix.bin, "rsmiBindings.py"))
+
+    @run_after("install")
+    def cache_test_sources(self):
+        """Copy the tests source files after the package is installed to an
+        install test subdirectory for use during `spack test run`."""
+        if self.spec.satisfies("@:5.1.0"):
+            return
+        self.cache_extra_test_sources([self.test_src_dir])
+
+    def test(self):
+        if self.spec.satisfies("@:5.1.0"):
+            print("Skipping: stand-alone tests")
+            return
+        test_dir = join_path(self.test_suite.current_test_cache_dir, self.test_src_dir)
+        with working_dir(test_dir, create=True):
+            cmake_bin = join_path(self.spec["cmake"].prefix.bin, "cmake")
+            prefixes = ";".join(
+                [
+                    self.spec["rocm-smi-lib"].prefix,
+                ]
+            )
+            rocm_smi_lib_path = ";".join([self.spec["rocm-smi-lib"].prefix.lib])
+            cc_options = [
+                "-DCMAKE_PREFIX_PATH=" + prefixes,
+                "-DRSMI_LIB_DIR=" + rocm_smi_lib_path,
```

Comment:
```suggestion
                "-DROCM_DIR=" + self.spec["rocm-smi-lib"].prefix,
```
I don't think it's necessary to set RSMI_LIB_DIR since it will already be set in the cmake file but we do need to set ROCM_DIR